app/security_system.py

```diff
@@ -66,13 +67,43 @@ def recognize_face(self, frame):
 
         # TODO 3
         # Perform face detection
+        faces = face_cascade.detectMultiScale(gray_frame, scaleFactor = 1.2, minNeighbors = 5, minSize = (30,30))
         # Iterate through each detected face
-        # Extract the detected face region
-        # Perform facial recognition on the detected face
-        # Draw bounding box around the detected face
-        # Annotate the frame with the recognized user's name and authorization status
-        # Log access attempt
-        # Your code goes here
+        for face in faces:
+            # Extract the detected face region
+            x,y,w,h = face
+            X,Y = x+w,y+h
+            image = gray_frame[y:Y, x:X]
+            # Perform facial recognition on the detected face
+            label, confidence = self.face_recognizer.predict(image)
+            person_name = self.get_person_name(label,confidence)
+            authorization_status = self.authorized_persons[person_name]
+            if(authorization_status==True):
+                color = (0,200,100)
+                is_authorized = "YES"
+            else:
+                color = (0,35,200)
+                is_authorized = "NO"
+            # Draw bounding box around the detected face
+            l= 20
+            t= 5
+            
+            cv2.rectangle(frame,face,color,1)
+            cv2.line(frame,(x,y),(x+l,y),color,t)
+            cv2.line(frame,(x,y),(x,y+l),color,t)
+            cv2.line(frame,(X-l,y),(X,y),color,t)
+            cv2.line(frame,(X,y),(X,y+l),color,t)
+            cv2.line(frame,(x,Y-l),(x,Y),color,t)
+            cv2.line(frame,(x,Y),(x+l,Y),color,t)
+            cv2.line(frame,(X-l,Y),(X,Y),color,t)
+            cv2.line(frame,(X,Y-l),(X,Y),color,t)
+            # Annotate the frame with the recognized user's name and authorization status
+           
+            cv2.rectangle(frame, [x,y-35,w-50,30], color,-1)
+            cv2.putText(frame,f"Name: {person_name}",(x+5,y-20),cv2.FONT_HERSHEY_PLAIN,0.8,(30,30,30),1)
+            cv2.putText(frame,f"Authorization Status: {authorization_status}",(x+5,y-10),cv2.FONT_HERSHEY_PLAIN,0.8,(0,0,0),1)
+            # Log access attempt
+            self.log_access_attempt(person_name,is_authorized)
```

Comment:
Here are our sample logs:
```
2023-10-05 19:06:10,989 - Person: Unknown, Authorized: False
2023-10-05 19:06:11,055 - Person: Madonna, Authorized: True
2023-10-05 19:06:11,120 - Person: Elton John, Authorized: True
```

We want to put True/False instead of Yes/No there.